### PR TITLE
Allows subscription with headers

### PR DESCRIPTION
**Pull request by ebenoist**
This reintroduces a ```SubscribeWithHeaders``` function to allow for arbitrary headers to be set when subscribing.

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -488,18 +488,27 @@ func (c *Conn) sendFrameWithReceipt(f *Frame) error {
 // will be received by this subscription. A subscription has a channel
 // on which the calling program can receive messages.
 func (c *Conn) Subscribe(destination string, ack AckMode) (*Subscription, error) {
+	return c.SubscribeWithHeaders(destination, ack, NewHeader())
+}
+
+func (c *Conn) SubscribeWithHeaders(destination string, ack AckMode, userDefined *Header) (*Subscription, error) {
```
Comment (gmassanek) [inline conn.go]: Nit: `userDefined` might not actually be defined by the user (eg: if invoked via `Subscribe`). I'd suggest naming something more descriptive like `headers` or `inputHeaders`?

**Comment by gmassanek** [comment]
LGTM

**Comment by ebenoist** [comment]
@gmassanek I couldn't agree more, but I wanted to match the pattern established in the Publish function's signature. @jjeffery thoughts?

**Comment by jjeffery** [comment]
Looks fine to me. I have been thinking of changing the API to make use of Dave Cheney's "Functional Options" pattern so that we can have just one function for subscribing with various different options. This would eliminate the need for SubscribeWithXXX functions. See http://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis for details. This would be a breaking change to the API though.

**Comment by ebenoist** [comment]
@jjeffery I recently read the same article and I couldn't agree more. Thank you for taking this in and thank you for you your efforts thus far. This is by far the best STOMP library available for Go.

**Pull request merged**